src/Symfony/Component/HttpClient/Response/CurlResponse.php

```diff
@@ -176,17 +173,35 @@ public function getInfo(string $type = null)
         return null !== $type ? $info[$type] ?? null : $info;
     }
 
+    /**
+     * {@inheritdoc}
+     */
+    public function getContent(bool $throw = true): string
+    {
+        $performing = self::$performing;
+        self::$performing = $performing || '_0' === curl_getinfo($this->handle, CURLINFO_PRIVATE);
```

Comment:
Performing here while the content is already available also kills performances when processing a large number of requests.